From f3a595e7efa76cf96cd43a47812d444f59d270cd Mon Sep 17 00:00:00 2001
From: "Christopher N. Hesse" <raymanfx@gmail.com>
Date: Wed, 26 Apr 2017 18:21:39 +0200
Subject: [PATCH 29/67] FaceDetect: Catch linker errors during initialization

Right now we only check if we can load the native
functions from the JNI, but unsatisfied linker errors
can still occur even if the lib is present.

Change-Id: Id2ac36374eb8341b278949d120dc7674d6d62691
(cherry picked from commit c44c4c3555bc7eadb0f0d35cd082ed90ef5731e2)
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 .../hz/selfportrait/detect/FaceDetect.java       | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/src/com/thundersoft/hz/selfportrait/detect/FaceDetect.java b/src/com/thundersoft/hz/selfportrait/detect/FaceDetect.java
index 90ccc59dc..c39c4ee70 100644
--- a/src/com/thundersoft/hz/selfportrait/detect/FaceDetect.java
+++ b/src/com/thundersoft/hz/selfportrait/detect/FaceDetect.java
@@ -42,7 +42,12 @@ public class FaceDetect {
      * initialize method,MUST called at first time.
      */
     public void initialize() {
-        mHandle = native_create();
+        try {
+            mHandle = native_create();
+        } catch (UnsatisfiedLinkError e) {
+            e.printStackTrace();
+            Log.e(TAG, "could not link native handle for ts_detected_face_jni library!");
+        }
     }
 
     public boolean isLibLoaded() {
@@ -54,7 +59,9 @@ public class FaceDetect {
      */
 
     public void uninitialize() {
-        native_destroy(mHandle);
+        if (mHandle != 0) {
+            native_destroy(mHandle);
+        }
     }
 
     /**
@@ -65,6 +72,11 @@ public class FaceDetect {
      * @return FaceInfo array if success, otherwise return null.
      */
     public FaceInfo[] dectectFeatures(Bitmap bmp) {
+        // check if the initialization failed
+        if (mHandle == 0) {
+            return null;
+        }
+
         int count = native_detect(mHandle, bmp);
         if (count < 1) {
             return null;
-- 
2.17.1

